jwt_proxy/api.py

```diff
@@ -2,12 +2,41 @@
 import jwt
 import requests
 import json
+import re
 
 from jwt_proxy.audit import audit_HAPI_change
 
 blueprint = Blueprint('auth', __name__)
 SUPPORTED_METHODS = ('GET', 'POST', 'PUT', 'DELETE', 'OPTIONS')
 
+# TODO: to be pulled into its own module and loaded per config
+def scope_filter(req, token):
+    # Check path
+    resource_pattern = rf"(Patient|DocumentReference)$"
+    if re.search(resource_pattern, req.path) is None:
+        return False
+    
+    user_id = token.get("sub")
+    identifier_pattern = rf"(https(:|%3[Aa])(\/|%2[Ff]){2}keycloak\.ltt\.cirg\.uw\.edu(%7[Cc]|\|))?{user_id}"
+    # Search params for keycloak id
+    params = req.args
+    id_param_value = params.get("identifier", params.get("_identifier", params.get("subject.identifier")))
+    if id_param_value is not None and re.search(identifier_pattern, id_param_value):
+        return True
+    # Search body for keycloak id
+    if req.is_json:
```

Comment:
the `request.json` property is a nice shortcut.  will be `None` if not included or wrong content-type.

Reply:
Thanks! [The Flask docs](https://tedboy.github.io/flask/interface_api.incoming_request_data.html#flask.Request.json) recommended preferring `get_json()`, though it looks like I don't need this check as [get_json() includes it implicitly](https://tedboy.github.io/flask/_modules/flask/wrappers.html#Request.get_json).